server/bleep/src/webserver/semantic.rs

```diff
@@ -1,84 +1,40 @@
 use super::prelude::*;
-use crate::{query::parser, semantic::Semantic};
+use crate::{
+    query::{
+        execute::ApiQuery,
+        parser::{self, ParsedQuery},
+    },
+    semantic::{self, Semantic},
+};
 use tracing::error;
 
-use qdrant_client::qdrant::value::Kind;
-use std::collections::HashMap;
-
-#[derive(Deserialize)]
-pub(super) struct Args {
-    limit: u64,
-    query: String,
-}
-
-#[derive(Serialize)]
-pub(super) struct SemanticResponse {
-    chunks: Vec<serde_json::Value>,
-}
-
-impl super::ApiResponse for SemanticResponse {}
-
 /// Get details of an indexed repository based on their id
 //
-#[utoipa::path(get, path = "/repos/indexed/:ref",
-    responses(
-        (status = 200, description = "Execute query successfully", body = SemanticResponse),
-        (status = 400, description = "Bad request", body = EndpointError),
-        (status = 500, description = "Server error", body = EndpointError),
-    ),
-)]
-pub(super) async fn raw_chunks(
-    Query(args): Query<Args>,
+pub(super) async fn complex_search(
```

Comment:
Why do we need this? Shouldn't this just be the behaviour that `/q` exposes?

Reply:
I'd want to transition this to `/q` after it stabilizes, but it's useful to have them compared side-by-side until that happens.